services/queue/src/artifacts.js

```diff
@@ -245,10 +246,12 @@ builder.declare({
     }
 
     // Update expiration and detail, which may have been modified
-    await artifact.modify((artifact) => {
-      artifact.expires = expires;
-      artifact.details = details;
-    });
+    artifact = artifactUtils.fromDbRows(
+      await this.db.fns.update_queue_artifact(
+        artifact.taskId, artifact.runId, artifact.name, details,
+        expires, artifact.etag
+      )
+    );
```

Comment:
Since this passes `etag`, it should handle conflicts.  I'm not sure it needs to handle the etag, though?  I think this is just for idempotency so we can assume that this call will override a previous call.

And, with that change, you can drop etag support for `update_queue_artifact`, which simplifies its implementation quite a bit.